libvips/foreign/vipspng.c

```diff
@@ -877,12 +881,110 @@ write_png_block( VipsRegion *region, VipsRect *area, void *a )
 	return( 0 );
 }
 
+#ifdef HAVE_IMAGEQUANT
+static int
+quantize_image( VipsImage *in, VipsImage *out, VipsImage *palette_out,
+	int colors, int Q, double dither )
+{
+	/* Ensure input is sRGB. */
+	if( in->Type != VIPS_INTERPRETATION_sRGB) {
+		VipsImage *srgb;
+		if( vips_colourspace( in, &srgb, VIPS_INTERPRETATION_sRGB,
```

Comment:
How about adding a test for CMYK and throwing an error? Otherwise colourspace will fail with a mysterious message.

Reply:
... if pngsave is asked to quanize an CMYK image, I mean.

Reply:
What I really want to do here is go from ushort to uchar and grayscale (1/2) bands to uchar with 3/4 bands in the most efficient way. If there is a better way to do that I am happy to chage it.

CMYK should already be blocked by an assertion further up the call stack in the pngsave case.

Reply:
Duh of course you're right, I was thinking of jpg. Carry on!